gatsby-node.js

```diff
@@ -1,31 +1,175 @@
 const createSlug = require('./src/util/createSlug');
+const path = require('path');
 
-exports.onCreatePage = async ({ page, actions }) => {
-  const { createPage, deletePage } = actions
+exports.createPages = ({ graphql, actions }) => {
+  const { createPage, createRedirect, createNodeField } = actions;
 
-  // If this is the learn page, accept all following paths.
-  console.log(page.path);
-  if (!!~page.path.indexOf('/learn')) {
-    deletePage(page);
-    createPage({
-      ...page,
-      path: '/'
-    });
-    createPage({
-      ...page,
-      matchPath: "/learn/*",
-    });
-  }
-}
+  // TODO: remove this redirect when we have an /index page
+  createRedirect({
+    fromPath: '/',
+    toPath: '/learn/introduction-to-nodejs',
+    redirectInBrowser: true,
+    force: true,
+  });
+
+  // TOOD: remove this redirect when we have a /learn index page
+  createRedirect({
+    fromPath: '/learn/',
+    toPath: '/learn/introduction-to-nodejs',
+    redirectInBrowser: true,
+  });
+
+  // TOOD: remove this redirect when we have a /learn index page
+  createRedirect({
+    fromPath: '/learn',
+    toPath: '/learn/introduction-to-nodejs',
+    redirectInBrowser: true,
+  });
+
+  return new Promise((resolve, reject) => {
+    const docTemplate = path.resolve('./src/templates/learn.tsx');
+    const filters = `filter: { fields: { slug: { ne: "" } } }`;
+
+    resolve(
+      graphql(
+        `
+          {
+            allMarkdownRemark(
+              ` +
+          filters +
+          `
```

Comment:
Hm. Interesting on my first attempt string templates didn't work. I assumed because it's graphql query but now it does :]

We need filters to sort out not necessary edges.

Reply:
Great! 👍 

And yup, I understand why we need filters. I wasn't suggesting to get rid of them altogether.

I'm suggesting to use it directly in the query instead of through a variable.

```ts
            allMarkdownRemark(
			  filter: { fields: { slug: { ne: "" } } },
              sort: { fields: [fileAbsolutePath], order: ASC }
            )
```

Reply:
Yep, that one we can do.